app/gui/src/dashboard/hooks/dragDelayHooks.ts

```diff
@@ -0,0 +1,85 @@
+/** @file Hooks to trigger an action on drag delay. */
+import type { DropEnterEvent, DropOptions } from '#/components/aria'
+import { useEventCallback } from '#/hooks/eventCallbackHooks'
+import type { DOMAttributes, DragEvent } from 'react'
+import { useRef } from 'react'
+
+/** The default delay, in milliseconds, before the drag action triggers. */
+const DEFAULT_DELAY_MS = 500
+
+/** Options for {@link useDragDelayAction}. */
+export interface DragDelayActionOptions {
+  readonly delayMs?: number
+}
+
+/** A callback for {@link useDragDelayAction}. */
+export type DragDelayCallback<T> = (event: DragEvent<T>) => void
+
+/** Trigger an action on a native HTML drop target. */
+export function useDragDelayAction<T>(
+  callback: DragDelayCallback<T> | undefined,
+  options: DragDelayActionOptions = {},
+) {
+  const { delayMs = DEFAULT_DELAY_MS } = options
+
+  const handle = useRef(0)
+
+  const cancelPreviousCallback = useEventCallback(() => {
+    clearTimeout(handle.current)
+    handle.current = 0
```

Comment:
I assume it should be null by default, WDYT? 

Also, worth noting that we have `useTimeout` hook: https://github.com/enso-org/enso/blob/develop/app/gui/src/dashboard/hooks/timeoutHooks.ts

Reply:
> `useTimeout` hook

just looked into it - not sure we want to subscribe to the state here, plus `useTimeoutApi` looks nice but unfortunately we need the `event` from the drag event so we can't use a static function unfortunately

... actually we could use a ref to hold the `event` maybe, but not sure if that's a good idea